### PR TITLE
Data type lessons

### PR DESCRIPTION
**Pull request by nkorinek**
Starting the data type lesson for week 2 of edsc

### CODE REVIEW

**Comment by nkorinek** [comment]
@lwasser @jlpalomino Hey! So I added a very tentative version of the vector lesson. I know we are planning on going over geopandas more in depth later in this course, so I changed the lesson to have less about the functionality of geopandas, aka using plot with matplotlib. I also took out a few cells I thought might go over the students head, like utilizing a for loop to access metadata. All in all, however, I kept MOST of the content from the original lesson. Not sure if I should have took out more, or if you all wanted to keep some sections I cut in. 

I added a few interactive activities, just looking at the basics of vector data attributes in geopandas. Obviously very open to notes, but it's at least started!

**Comment by lwasser** [comment]
i'll have a look @nkorinek this is a perfect end of the day thing for me to look at :) thank you. i may make edits so im going to assume you are done working on it for today?

**Comment by nkorinek** [comment]
@lwasser No I'll keep working for about an hour today! Getting a bit of a head start on the raster lesson

**Comment by lwasser** [comment]
oops @nkorinek i went ahead and left notes in the vector notebook. i assumed because you pinged me you were not working on that lesson anymore! if it causes big conflicts let me me know.

So a few things

1. you have all of the background on spatial data in there. i think we can provide the clif notes summary - skip most of the metadata discussion and refer to the reading that has more information about metadata - we will cover that in the spatial workshop in 2 weeks. 
2. for this lesson let's have them get to creating some maps quickly. I suggest that we download natural earth data and i've put some links into the notebook. maybe a challenge at the end could be showing them how to crop data using a boundary - subset out a county or state and crop something else using the geopandas clip function ... that would be a nice challenge. 
3. again all of the activities are metadata focused which won't be exciting for students at the beginning. let's start with making maps of things. feel free to explore other natural earth layers but i think they will work out well. i've cut down some of the text and left notes in the notebook about what i think would be good for this one. let me know if you have questions.

How about this for the future - when you push something up and ping me i am going to assume it's ready for a review which means you are not working on that file anymore. so please don't ping me until you are at a stopping point for that file! it's cool if you continue to work on other files but for the file you ask me to review, i will assume you have stopped working on that specific file and i can edit it directly. does that sound like a plan?

**Comment by lwasser** [comment]
@nkorinek one more thing before i forget !! each of these notebooks should showcase a few different types of data that are often available in this format. So you can use natural earth -- and then maybe ask jenny or find some small downloadable other datasets that contains other types of vector data as demos. so thtere is likely a section on this page like "What types of data are often in spatial vector format" or something like that. you can have a short list including

* census
* boundaries - roads, contries, water bodies etc
* other summarized ata maybe like precipitation (by county or area), etc - just a list of some examples - we can help.

then the lesson might showcase a few different kinds - showing a few plots that the students maybe modify and then an activity at the end (maybe with natural earth) where you say -- now make this map of x, and x kind of thing.

the metata stuff can be a breakout with links to the intermediate lessons. 
we can chat more monday but again this is an overview page with a focus on different types of vector (or raster, etc) data showcasing plots and mapping in the interactive activities and demo examples!

**Comment by nkorinek** [comment]
@lwasser alright, I've gone through and tried to implement your suggestions for the vector data lesson. The edits made sense, and I hope that I did it somewhat like how you envisioned. Open to any feedback. I still need to write the tests for the new interactive activities, but you should be able to review this notebook/make edits without my test writing interrupting.

**Comment by lwasser** [comment]
@nkorinek UPDATE: i'm about 1/2 maybe more through the vector lesson edits. i suggest not working on it until i'm done as i'm doing a fairly significant overhaul of it. i think we should create a second activity using census data if there is a dataset we can use somewhere. to demo a second kind of data. anyway more later. i started work at 7am today so i could run when it wasn't raining (now!!). i'll be back in a bit. (~1.5 hours)

File: data-types/02-intro-vector-data.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+# fig, ax = plt.subplots()
+#
+# countries.boundary.plot(ax=ax)
+# your_dataframe_here.plot(ax=ax, color='blue');
+# ax.set(title="Put Title Here");
+# ```
+
+
+
+# <div class="notice--warning" markdown="1">
```
Comment (lwasser) [inline data-types/02-intro-vector-data.py]: NOTE - i ran out of steam here! Feel free to edit everywhere but i know this section needs a bit more work.

**Comment by jlpalomino** [comment]
@lwasser I reviewed the tabular notebook and it is looking good!  I left your to-dos, etc, as unfortunately, I do not have time at this moment for more than an overall review. 

I fixed a few typos and added hyperlinks in a few places where you highlighted a need. Other than that, no major comments or suggestions, as I think the activity are coming together nicely.  I will look at the vector notebook next.

**Comment by jlpalomino** [comment]
@lwasser The vector notebook is also looking good.  Again, fixed a few typos and added a few links, etc.  Overall, I really like the activities.

I do wonder about the last bonus activity on clipping though. I understand it is intended for more advanced students but it may discourage the others who reach that point successfully and thought they were getting a hang of it all?  Just a thought.

**Comment by lwasser** [comment]
@jlpalomino sure - tell me more? do you think it will be hard for them to understand what clipping does? i wanted to provide one simple geoprocessing demo in the notebook. open to any suggestions. we just though clip was simple enough as it's just one line of code and we give them the code ... but i guess there is a subset in there too.. let me know! 

i've finalized the tabular notebook! and can work on this next.

**Comment by nkorinek** [comment]
@lwasser @jlpalomino sorry to add more to your plates, but I've pushed up the first draft of the raster lesson. That can be reviewed whenever you get the chance.

**Comment by lwasser** [comment]
ok raster lesson is in! @nkorinek i am going to walk my dog. i'll look at your final activity after that!  thank you for all of the help today!

**Comment by nkorinek** [comment]
@lwasser sounds good, exercise lesson has been pushed up! If you have any questions about it you can reach me on slack

File: data-types/04-data-types-exercise.ipynb [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"notice--warning\" markdown=\"1\">\n",
+    "\n",
+    "## <i class=\"fa fa-pencil-square-o\" aria-hidden=\"true\"></i> Challenge: Open a Text File\n",
+    "\n",
+    "Below there is a cell of imports that will help you with the challenges. The first challenge is to download this csv file containing information on the climbing formations in the boulder area here: \n",
+    "\n",
+    "https://opendata.arcgis.com/datasets/175425c25d8849b58feb89483ef02961_1.csv\n",
```
Comment (lwasser) [inline data-types/04-data-types-exercise.ipynb]: @nkorinek did this download work for you? this is not working for me.

**Pull request merged**